packages/workspace/src/tasks-runner/task-orderer.ts

```diff
@@ -18,26 +24,84 @@ export class TaskOrderer {
     )
       return [tasks];
     if (tasks.length === 0) return [];
-    const res = [];
+    const stages: Task[][] = [];
```

Comment:
It looks like the complexity of this function is O(N^2). Which is the same as in master. The one in master could be made O(N). I'm curious to see how long it takes for a large repo with 1000 nodes.